docs/reference/api/docker_remote_api_v1.24.md

```diff
@@ -461,6 +462,20 @@ Json Parameters:
     -   **CgroupParent** - Path to `cgroups` under which the container's `cgroup` is created. If the path is not absolute, the path is considered to be relative to the `cgroups` path of the init process. Cgroups are created if they do not already exist.
     -   **VolumeDriver** - Driver that this container users to mount volumes.
     -   **ShmSize** - Size of `/dev/shm` in bytes. The size must be greater than 0.  If omitted the system uses 64MB.
+    -   **Mounts** - List of mount configurations to create the container with. This replaces `Binds` and `Volumes`. If `Mounts` is specified along with either `Binds` or `Volumes`, the request will be rejected. Fields for a mount configuration:
+          + **Type** - String specifying the type of mount, e.g. `hostbind`, `ephemeral`, or `persistent` (required)
+          + **Source** - String specifying the host path to use for the mount. Path must exist. (required, `hostbind` only)
+          + **Destination** - String specifying the container destination path for the mount (required)
+          + **Driver** - String specifying the driver to use when creating a volume (required, `persistent` only)
+          + **Name** - String specifying the name of the volume to create (required, `persistent` only)
+          + **Mode** - comma sepparated list of mount modes, e.g. `ro`, `rw`, `nocopy`
+          + **CreateOpts** - A map of key/value pairs (strings) to use when creating the volume, passed directly to the volume driver. (`persistent only)
+
+          Mount types:
+            + **hostbind** - Mounts a file or directory from the host into the container. Must exist prior to creating the container.
```

Comment:
I'm not sure what you mean by `names` here. Could you give an example?

I would think that we would both have a way to configure number of containers that can mount the volume and reference counted retention policy. If the volume can be mounted by multiple containers and the retention policy is to remove it after use, the volume would be removed when there are no more users.

This approach does create a slight problem: how do you create a volume that is to be removed after use before being used? This can be handled by special casing "new" volumes.